Month 7/step 2

Aldukhov opened the pull request.

Aldukhov committed:
CRA.1.1

Aldukhov committed:
Fix components

change to function comp.
add scrollInto effect
create folders for every components

Aldukhov committed:
Final JSX + layout

add

-api
-modal
-propTypes

kyzinatra commented on src/components/burger/ingredients/Food-items/food-items.jsx:
```diff
@@ -0,0 +1,93 @@
+            {isModalOpen && (
+                <ModalOverlay onClick = {closeModal}>
+                    <Modal onClose ={closeModal} details={'ingridients'}>
+                        <IngredientDetails onClose ={closeModal} description={selectedItem}/>
```
Надо исправить: `ModalOverlay` должна быть частю `<Modal />` мы не должны думать о том, что нам нужна подложка под модальным окном

kyzinatra commented on src/components/ModalOverlay/modalOverlay.jsx:
```diff
@@ -0,0 +1,46 @@
+    const handleEscClose = (evt) => {
+        if ((evt.key === 'Escape')) {
+            onClick();
+          }
```
Надо исправить: Обработка `escape` должна проиходить в компоненте `<Modal>`, так как это просто подложка

kyzinatra commented on src/components/ModalOverlay/modalOverlay.jsx:
```diff
@@ -0,0 +1,46 @@
+        return() => {
+            document.removeEventListener('click',handleOverplayClose)
+            document.removeEventListener('keydown',handleEscClose)
+        }
```
Отлично! Круто, что вы очищаете обработчики

kyzinatra commented on src/components/ModalOverlay/modalOverlay.jsx:
```diff
@@ -0,0 +1,46 @@
+        (
+            <div className={styles.overlay}>
+                {children}
+            </div>
```
Можно лучше: не стоит оборачивать в подложку контент модального окна. Так могут возникать некоторые баги. Лучше, чтобы она была рядом.

```js
return (
  <ModalOverlay /> 
  ... // остальное модальное окно
)
```

kyzinatra commented on src/components/ModalOverlay/modalOverlay.jsx:
```diff
@@ -0,0 +1,46 @@
+    return ReactDOM.createPortal(
+        (
+            <div className={styles.overlay}>
+                {children}
```
Надо исправить: если кликнуть на контент модального окна оно закроется. Так быть не должно

kyzinatra commented on src/components/burger/constructor/constructor.jsx:
```diff
@@ -0,0 +1,22 @@
+
+BurgerConstructor.protoTypes = {
+    data: PropTypes.object
+}
```
Надо исправить: необходимо описать все типы. Если это обьект, то нужно указать какие именно полям есть в этом объекте. Это можно сделать с помощью `.shape()`

kyzinatra commented on src/components/burger/ingredients/Categories/categories.jsx:
```diff
@@ -0,0 +1,26 @@
+Categories.protoTypes = {
+  scrollBun: PropTypes.any,
+  scrollSauce: PropTypes.any,
+  scrollMain: PropTypes.any
```
Надо исправить: по аналогии необходимо использовать точные типы и не использовать `any`

kyzinatra commented on src/components/burger/ingredients/Food-items/food-items.jsx:
```diff
@@ -0,0 +1,93 @@
+        let items = filterType.map((element) => {
+            return (
+                <div className={classNames(styles.item)} key={element.id} onClick={() => openModal(element)}>
+                    <img className={classNames(styles.item__image, 'ml-3 mr-3')} src={element.image} alt={element.name} />
```
Можно лучше: это выглядит как отдельный компонент. Можно его так и вынести

kyzinatra commented on src/components/burger/constructor/Info/info.jsx:
```diff
@@ -0,0 +1,44 @@
+
+            {isModalOpen && (
+                <ModalOverlay onClick = {closeModal}>
+                    <Modal onClose ={closeModal} details={'order'}>
```
Отлично! Здорово, что вы удаляете модальное окно когда оно закрыто

kyzinatra commented:
Выглядит очень круто! Видно, что проделана большая работа. Но все же есть моменты на которые стоит обратить внимание.
Подробнее: 

**Что важно исправить:**

-  `ModalOverlay` должна быть частью `<Modal />` мы не должны думать о том, что нам нужна подложка под модальным окном
-  Обработка `escape` должна происходить в компоненте `<Modal>`, так как это просто подложка
-  если кликнуть на контент модального окна оно закроется. Так быть не должно
-  необходимо описать все типы. Если это объект, то нужно указать какие именно полям есть в этом объекте. Это можно сделать с помощью `.shape()`
-  по аналогии необходимо использовать точные типы и не использовать `any`


**Можно лучше:**

-  не стоит оборачивать в подложку контент модального окна. Так могут возникать некоторые баги. Лучше, чтобы она была рядом.

```js
return (
  <ModalOverlay /> 
  ... // остальное модальное окно
)
```
-  это выглядит как отдельный компонент. Можно его так и вынести


**Что понравилось:**

-  Круто, что вы очищаете обработчики
-  Здорово, что вы удаляете модальное окно когда оно закрыто


**Обратите внимание, что работа принимается только после исправления всех `«Надо исправить»`.**

Aldukhov committed:
FIx problems

- change PropTypes
- remake "handleOverplayClose"
- remake Modal and ModalOverplay
